compiler/src/org.graalvm.compiler.core.test/src/org/graalvm/compiler/core/test/GraalOptimizationTest.java

```diff
@@ -69,4 +87,9 @@ protected void checkLIR(String methodName, Predicate<LIRInstruction> predicate,
         Assert.assertEquals(expected, actualOpNum);
     }
 
+    protected void checkNodeCount(String methodName, Class<? extends Node> expectedClass, int expected) {
+        compile(getResolvedJavaMethod(methodName), null);
+        int actual = structuredGraph.getNodes().filter(expectedClass).count();
+        Assert.assertEquals(expected, actual);
+    }
```

Comment:
I will move this function and its related codes to "IfNodeCanonicalizationTest2.java" and revert all the modifications in this file, to keep this file used for backend optimizations alone.